From b0f71e5c05972314bf5a7318019e600b2b8d9daf Mon Sep 17 00:00:00 2001
From: Wouter Wijsman <wwijsman@live.nl>
Date: Wed, 11 Mar 2020 23:09:21 +0100
Subject: [PATCH] Allow installing in /usr/local

With this commit relative paths will be used, making users able to
install with pip and setuptools without requiring packages.

Fixes #85
---
 minigalaxy/paths.py | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/minigalaxy/paths.py b/minigalaxy/paths.py
index 6503f0f..6e2a79c 100644
--- a/minigalaxy/paths.py
+++ b/minigalaxy/paths.py
@@ -1,4 +1,7 @@
 import os
+import sys
+
+LAUNCH_DIR = os.path.abspath(os.path.dirname(sys.argv[0]))
 
 CONFIG_DIR = os.path.join(os.getenv('XDG_CONFIG_HOME', os.path.expanduser('~/.config')), "minigalaxy")
 CONFIG_FILE_PATH = os.path.join(CONFIG_DIR, "config.json")
@@ -7,14 +10,14 @@
 THUMBNAIL_DIR = os.path.join(CACHE_DIR, "thumbnails")
 DEFAULT_INSTALL_DIR = os.path.expanduser("~/GOG Games")
 
-UI_DIR = "data/ui"
+UI_DIR = os.path.abspath(os.path.join(LAUNCH_DIR, "../data/ui"))
 if not os.path.exists(UI_DIR):
-    UI_DIR = "/usr/share/minigalaxy/ui"
+    UI_DIR = os.path.abspath(os.path.join(LAUNCH_DIR, "../share/minigalaxy/ui"))
 
-LOGO_IMAGE_PATH = "data/minigalaxy.png"
+LOGO_IMAGE_PATH = os.path.abspath(os.path.join(LAUNCH_DIR, "../data/minigalaxy.png"))
 if not os.path.exists(LOGO_IMAGE_PATH):
-    LOGO_IMAGE_PATH = "/usr/share/icons/hicolor/192x192/apps/minigalaxy.png"
+    LOGO_IMAGE_PATH = os.path.abspath(os.path.join(LAUNCH_DIR, "../share/icons/hicolor/192x192/apps/minigalaxy.png"))
 
-LOCALE_DIR = "data/mo"
+LOCALE_DIR = os.path.abspath(os.path.join(LAUNCH_DIR, "../data/mo"))
 if not os.path.exists(LOCALE_DIR):
-    LOCALE_DIR = "/usr/share/minigalaxy/translations"
+    LOCALE_DIR = os.path.abspath(os.path.join(LAUNCH_DIR, "../share/minigalaxy/translations"))
